components/paging/pageable-mixin.js

```diff
@@ -1,35 +1,67 @@
 import { html } from 'lit';
+import { SubscriberRegistryController } from '../../controllers/subscriber/subscriberControllers.js';
 
 export const PageableMixin = superclass => class extends superclass {
 
+	static get properties() {
+		return {
+			/**
+			 * Total number of items. If not specified, neither it nor the count of items showing will be displayed.
+			 * @type {number}
+			 */
+			itemCount: { type: Number, attribute: 'item-count', reflect: true },
+			_itemShowingCount: { state: true },
+		};
+	}
+
 	constructor() {
 		super();
-		this._pageable = true;
+
+		this.itemCount = -1;
+		this._itemShowingCount = 0;
+		this._pageableSubscriberRegistry = new SubscriberRegistryController(this, 'pageable', {
+			onSubscribe: this._updatePageableSubscriber.bind(this),
+			updateSubscribers: this._updatePageableSubscribers.bind(this)
+		});
 	}
 
-	/* must be implemented by consumer */
-	_getItemByIndex(index) { } // eslint-disable-line no-unused-vars
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._updateItemShowingCount();
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		if (changedProperties.has('itemCount') || changedProperties.has('_itemShowingCount')) {
+			this._pageableSubscriberRegistry.updateSubscribers();
+		}
+	}
 
 	/* must be implemented by consumer */
-	async _getItemsShowingCount() { }
+	_getItemByIndex(index) { } // eslint-disable-line no-unused-vars
 
 	/* must be implemented by consumer */
-	_getLastItemIndex() { }
+	_getItemShowingCount() { }
 
-	async _handlePagerSlotChange(e) {
-		this._updatePagerCount(await this._getItemsShowingCount(), e.target);
+	_getLastItemIndex() {
+		return this._itemShowingCount - 1;
 	}
 
 	_renderPagerContainer() {
-		return html`<slot name="pager" @slotchange="${this._handlePagerSlotChange}"></slot>`;
+		return html`<slot name="pager"></slot>`;
 	}
 
-	_updatePagerCount(count, slot) {
-		if (!slot) slot = this.shadowRoot.querySelector('slot[name="pager"]');
-		const elements = slot.assignedElements({ flatten: true });
-		if (elements.length > 0) {
-			elements[0].itemShowingCount = count;
-		}
+	_updateItemShowingCount() {
+		this._itemShowingCount = this._getItemShowingCount();
+	}
```

Comment:
This essentially replaces `_updatePagerCount`, and is what consumers need to call any time the count changes. It's a bit different in that the subscribers will be updated in the next update cycle (via the `updated` check), rather than immediately.

If we were very concerned about that immediacy I could change `_itemShowingCount` to an internal property and explicitly call `updateSubscribers` from here - I'm okay with either approach, but the current way protects us a bit more in case of consumers explicitly setting `_itemShowingCount` for any reason.